executor/aggfuncs/func_max_min.go

```diff
@@ -14,49 +14,126 @@
 package aggfuncs
 
 import (
+	"container/heap"
+
 	"github.com/pingcap/tidb/sessionctx"
 	"github.com/pingcap/tidb/types"
 	"github.com/pingcap/tidb/types/json"
 	"github.com/pingcap/tidb/util/chunk"
 	"github.com/pingcap/tidb/util/stringutil"
 )
 
+type maxMinHeap struct {
+	data    []interface{}
+	h       heap.Interface
+	varSet  map[interface{}]int64
```

Comment:
The Decimal type may not be suitable for this map. We should treat `1.0` and `1.00` as the same value, but not on this map. `MyDecimal.ToHashKey` may help.